keylime/ima/ima.py

```diff
@@ -625,46 +462,69 @@ def ima_policy_db_contents(ima_policy_name: str, ima_policy: str, tpm_policy: st
     else:
         ima_policy_db_format["ima_policy"] = ima_policy
     ima_policy_json = json.loads(ima_policy)
-    if "allowlist" in ima_policy_json:
-        if "meta" in ima_policy_json["allowlist"]:
-            if "checksum" in ima_policy_json["allowlist"]["meta"]:
-                ima_policy_db_format["checksum"] = ima_policy_json["allowlist"]["meta"]["checksum"]
-            else:
-                alist_bytes = json.dumps(copy.deepcopy(EMPTY_ALLOWLIST)).encode()
-                sha256 = hashlib.sha256()
-                sha256.update(alist_bytes)
-                ima_policy_db_format["checksum"] = sha256.hexdigest()
+    if "meta" in ima_policy_json:
+        if "checksum" in ima_policy_json["meta"]:
+            ima_policy_db_format["checksum"] = ima_policy_json["meta"]["checksum"]
+        else:
+            alist_bytes = json.dumps(ima_policy).encode()
+            sha256 = hashlib.sha256()
+            sha256.update(alist_bytes)
+            ima_policy_db_format["checksum"] = sha256.hexdigest()
     if tpm_policy:
         ima_policy_db_format["tpm_policy"] = tpm_policy
     return ima_policy_db_format
 
 
-def read_excllist(exclude_path=None):
-    excl_list = []
-    if exclude_path and os.path.exists(exclude_path):
-        with open(exclude_path, encoding="utf-8") as f:
-            for line in f:
-                line = line.strip()
-                if line.startswith("#") or len(line) == 0:
-                    continue
-                excl_list.append(line)
+def verify_ima_policy(ima_policy, ima_policy_key=None, ima_policy_sig=None, verify_sig=True):
+    """
+    Verify that an IMA policy is valid. If provided IMA policy has a detached signature, verify the signature.
+    """
+
+    if ima_policy is None:
+        raise ImaValidationError("No IMA policy provided!", 400)
```

Comment:
For uniformity, `message="..", code=400`?

Reply:
Fixed, thanks!